runtime/compiler/optimizer/IdiomRecognition.cpp

```diff
@@ -1066,16 +1066,7 @@ TR_CISCGraph::initializeGraphs(TR::Compilation *c)
    bool genMemcmp = c->cg()->getSupportsArrayCmp();
    bool genIDiv2Mul = c->cg()->getSupportsLoweringConstIDiv();
    bool genLDiv2Mul = c->cg()->getSupportsLoweringConstLDiv();
-   // FIXME: We need getSupportsCountDecimalDigit() like interface
-   // this idiom is only enabled on 390 for the moment
-
-#if defined(J9VM_OPT_JITSERVER)
-   // Enabling genDecimal generates the TROT instruction on Z which is currently not
-   // relocatable for remote compiles. Thus we disable this option for remote compiles for now.
-   bool genDecimal = c->target().cpu.isZ() && !c->isOutOfProcessCompilation();
-#else
-   bool genDecimal = c->target().cpu.isZ();
-#endif /* defined(J9VM_OPT_JITSERVER) */
+   bool genDecimal = c->cg()->getSupportsCountDecimalDigit();
```

Comment:
This is the wrong place to avoid transforming a pattern under AOT. `initializeGraphs()` runs only once and allocates the graphs in persistent memory. By making this sensitive to `compileRelocatableCode()`, this global set of graphs will depend on whether the graphs happened to be created during an AOT or a non-AOT compilation.

If we create the graphs in an AOT compilation, then this will prevent the transformation in AOT compilations, but also (unnecessarily) in all other compilations. Worse, if we create the graphs in a non-AOT compilation, we'll include the digit counting graph, and then we might transform it in a later AOT compilation.

I think the check for out-of-process compilations is fine though, since I expect that at this point `isOutOfProcessCompilation()` is always true on the server, and always false on the client.

So we should generate the graph under exactly the same conditions as we used to, and then have the transformer function fail under AOT.

By the way, I disagree with this `FIXME` - I don't think OMR should need to be concerned with this

Reply:
I've closed the omr PR, and updated this PR to bail on the transformation in the specific transformer. Change here: d3b3c24ff870d91c8e14bdf2689da14077732f2f